Library/Homebrew/utils/shell.rb

```diff
@@ -89,7 +89,12 @@ def prepend_path_in_shell_profile(path)
     # cut the actual value of ARG_MAX in half for safety since
     # environment variables count toward the total
     def max_command_length
-      Utils.popen_read("/usr/bin/getconf", "ARG_MAX").to_i / 2
+      Utils.popen_read("/usr/bin/getconf", "ARG_MAX").to_i - 2 * environment_variables_length
```

Comment:
Done in d8a6863283048cd2cc1f7fb78178b2fc19420e0c. Much less code indeed! :smile: